### DISCUSSION TITLE
How to configure Gradle updates to use strict versions?

### DISCUSSION THREAD

**Discussion by Sarah-Gallitz** [renovatebot/renovate#8038] [Request Help] [answered]
I would like to configure gradle to use strict versions.

Currently, I specify
```
"androidx.browser:browser:1.0.0!!"
```

renovate updates it to
```
"androidx.browser:browser:1.3.0"
```

but I would like renovate to use
```
"androidx.browser:browser:1.3.0!!"
```

I read the docs for renovates gradle versioning, and it says `rangeStrategy: pin` is supported, so I enabled that, but it didn't seem to make any difference to the gradle updates (other package managers started pinning things though, so the setting did take effect).

https://docs.renovatebot.com/modules/versioning/#gradle-versioning

Anyone know if this is possible?

**Comment by viceice** [comment, marked as answer]
I think this is currently not supported. I've never seen those strict versions, do you have any reference for that?

**Comment by Sarah-Gallitz** [comment]
Sure do! It is the shorthand syntax for strict versioning.

Gradles definition of `strict` versions:
>strictly
Any version not matched by this version notation will be excluded. This is the strongest version declaration. On a declared dependency, a strictly can downgrade a version. When on a transitive dependency, it will cause dependency resolution to fail if no version acceptable by this clause can be selected. See overriding dependency version for details. This term supports dynamic versions.
>When defined, overrides previous require declaration and clears previous reject.

https://docs.gradle.org/current/userguide/rich_versions.html

Reference for the shorthand syntax:
```kt
dependencies {
    // short-hand notation with !!
    implementation("org.slf4j:slf4j-api:1.7.15!!")
    // is equivalent to
    implementation("org.slf4j:slf4j-api") {
        version {
           strictly("1.7.15")
        }
    }
}
```
https://docs.gradle.org/current/userguide/single_versions.html#simple_version_declaration_semantics

**Reply by viceice** [reply]
In this case please open a new feature request issue for adding support. you can reference this discussion there.

It would also be very helpful for an implementer to have a simple reproduction repo to test against.
